### DISCUSSION TITLE
Switch/Toggle guidance clarification

### DISCUSSION THREAD

**Discussion by mma1504** [twilio-labs/paste#3156] [General] [answered]
Hi Paste team! The general wisdom is use a toggle/switch when the change is instant, use a checkbox when it's not instant. More specifically, our Paste guidance says:

"A checkbox should not be used to apply a user’s choice instantly. In that case, such as enabling a system feature, you should consider using the Switch component."

In Segment, we have a pattern where in a source/destination settings page where you can enable or disable the source/destination. It's currently implemented as a toggle, but you have to press Save afterwards. On it's surface, it seems like we should be using a checkbox given Paste guidance, but the line about "such as enabling a system feature" seems to suggest the Switch is OK. 

What is the Paste team's perspective on this?

**Comment by SiTaggart** [comment, marked as answer]
Hi @mma1504 

I think if I applied our advice to the example in Segment from the opposite direction, we'd suggest a checkbox.

What I mean by that is, if you are not enabling a system feature instantly, you should be using a checkbox. The assumption in our wording is that feature enablement is and should be instant, and I can see how that can be confusing if there is a technical limitation to an existing implementation.

TL;DR - Instant = switch. Not instant = checkbox
